File: frontend/src/lib/health.test.ts
import { describe, expect, it } from 'vitest';
import { appURL, toStatusRows, type HealthReport } from './health';

describe('toStatusRows', () => {
  it('maps dependency health to stable status rows', () => {
    const report: HealthReport = {
      status: 'not_ready',
      service: 'opskeeper-api',
      version: 'test',
      timestamp: '2026-01-01T00:00:00Z',
      checks: {
        postgres: { status: 'up', latency_ms: 4 },
        redis: { status: 'down', latency_ms: 10 }
      }
    };

    expect(toStatusRows(report)).toEqual([
      { name: 'API', status: 'down' },
      { name: 'PostgreSQL', status: 'up', latency: '4 ms' },
      { name: 'Redis', status: 'down', latency: '10 ms' }
    ]);
  });
});

describe('appURL', () => {
  it('resolves application paths from the document base URI', () => {
    const base = document.createElement('base');
    base.href = 'http://localhost:5173/opskeeper/';
    document.head.append(base);

    expect(appURL('/api/v1/teams').toString()).toBe(
      'http://localhost:5173/opskeeper/api/v1/teams'
    );

    base.remove();
  });

  it('resolves application paths from a root base URI', () => {
    const base = document.createElement('base');
    base.href = 'http://localhost:5173/';
    document.head.append(base);

    expect(appURL('/api/v1/teams').toString()).toBe(
      'http://localhost:5173/api/v1/teams'
    );

    base.remove();
  });
});

File: backend/health/health_test.go
package health

import (
	"context"
	"errors"
	"testing"
	"time"
)

func TestReadinessReportsFailedDependency(t *testing.T) {
	service := NewService(time.Second, []Check{
		{Name: "postgres", Run: func(context.Context) error { return nil }},
		{Name: "redis", Run: func(context.Context) error { return errors.New("unavailable") }},
	})

	report := service.Readiness(context.Background(), "test")
	if report.Status != "not_ready" {
		t.Fatalf("Readiness() status = %q, want not_ready", report.Status)
	}
	if report.Checks["redis"].Status != "down" {
		t.Fatalf("Readiness() redis = %#v, want down", report.Checks["redis"])
	}
}
